src/CMakeLists.txt

```diff
@@ -18,10 +18,12 @@ else()
 endif()
 
 add_library(benchmark ${SOURCE_FILES} ${RE_FILES})
+find_package(Threads REQUIRED)
 
-# Link threading if building a shared library.
+# Link threading if building a shared library, otherwise let client do it.
 if (BUILD_SHARED_LIBS)
-  find_package(Threads REQUIRED)
+  target_link_libraries(benchmark PRIVATE ${CMAKE_THREAD_LIBS_INIT})
+else()
   target_link_libraries(benchmark ${CMAKE_THREAD_LIBS_INIT})
```

Comment:
Judging from the [gcc documentation](https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html#index-static-1225), gcc will link against shared libs when available, unless you pass `-static`. CMake doesn't seem to pass that flag, so I think this will not statically link pthread even when building static libs, and so it should work fine even without `INTERFACE`.